modules/pam_env/pam_env.c

```diff
@@ -625,34 +823,41 @@ static int
 _parse_config_file(pam_handle_t *pamh, int ctrl, const char *file)
 {
     int retval;
-    char buffer[BUF_SIZE];
-    FILE *conf;
     VAR Var, *var=&Var;
-
-    D(("Called."));
+    char **conf_list = NULL;
 
     var->name=NULL; var->defval=NULL; var->override=NULL;
 
-    D(("Config file name is: %s", file));
+    D(("Called."));
 
+#ifdef VENDOR_DEFAULT_CONF_FILE
+#ifndef USE_ECONF
+    /* Only one file will be parsed. So, file has to be set. */
+    if (file == NULL) /* No filename has been set via argv. */
+      file = DEFAULT_CONF_FILE;
     /*
-     * Lets try to open the config file, parse it and process
-     * any variables found.
-     */
-
-    if ((conf = fopen(file,"r")) == NULL) {
-      pam_syslog(pamh, LOG_ERR, "Unable to open config file: %s: %m", file);
-      return PAM_IGNORE;
+    * Check whether DEFAULT_CONF_FILE file is available.
+    * If it does not exist, fall back to VENDOR_DEFAULT_CONF_FILE file.
+    */
+    struct stat stat_buffer;
+    if (stat(file, &stat_buffer) != 0 && errno == ENOENT) {
+      file = VENDOR_DEFAULT_CONF_FILE;
     }
+#endif
+#endif
 
-    /* _pam_assemble_line will provide a complete line from the config file,
-     * with all comments removed and any escaped newlines fixed up
-     */
+#ifdef USE_ECONF
+    /* If "file" is not NULL, only this file will be parsed. */
+    retval = econf_read_file(pamh, file, " \t", PAM_ENV, ".conf", "security", &conf_list);
+#else
+    retval = read_file(pamh, file, &conf_list);
```

Comment:
I think the `#ifdef VENDOR_DEFAULT_CONF_FILE` hunk could be moved right before the `read_file` invocation, this would make `#ifndef USE_ECONF` there unneeded.

Reply:
Yes, good point